contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,417 @@
+            abi.encodePacked(byte(0x19), byte(0), this, token, to, amount, safeTxGas, dataGas, gasPrice, gasToken, refundReceiver, _nonce)
+        );
+    }
+
+    /// @dev Returns start of the current time period.
+    /// @return Unix timestamp.
+    function currentStartTime()
+        public
+        view
+        returns (uint)
+    {
+        return getNow() - (getNow() % timePeriod);
+    }
+
+    function handleTransferLimits(address token, uint256 amount)
+        internal
+        returns (bool)
+    {
+        TransferLimit storage transferLimit = transferLimits[token];
+        // If time period is over, reset expenditure.
+        if (isPeriodOver()) {
+            transferLimit.spent = 0;
+            totalWeiSpent = 0;
+            totalDaiSpent = 0;
+        }
+
+        // Transfer + previous expenditure shouldn't exceed limit specified for token.
+        if (SafeMath.add(transferLimit.spent, amount) > transferLimit.transferLimit) {
+            return false;
+        }
+
+        // If a global cap is set, transfer amount + value of all
+        // previous expenditures (for all tokens) shouldn't exceed global limit.
+        if (!isUnderGlobalCap(token, amount)) {
+            return false;
+        }
+
+        transferLimits[token].spent = SafeMath.add(transferLimits[token].spent, amount);
+
+        return true;
+    }
+
+    function isUnderGlobalCap(address token, uint256 amount)
+        internal
+        returns (bool)
+    {
+        if (globalWeiCap == 0 && globalDaiCap == 0) {
+            return true;
+        }
+
+        // Calculate value in ether.
+        uint256 ethNum;
+        uint256 ethDen;
+        (ethNum, ethDen) = getEthAmount(token, amount);
+
+        // Convert ether to wei
+        uint256 weiAmount = SafeMath.div(SafeMath.mul(ethNum, 10**18), ethDen);
+        if (globalWeiCap > 0 && SafeMath.add(totalWeiSpent, weiAmount) > globalWeiCap) {
+            return false;
+        }
+        totalWeiSpent = SafeMath.add(totalWeiSpent, weiAmount);
+
+        if (globalDaiCap != 0) {
+            // Calculate value in dai.
+            uint256 daiNum;
+            uint256 daiDen;
+            (daiNum, daiDen) = getDaiAmount(ethNum, ethDen);
+
+            uint256 daiAmount = SafeMath.div(daiNum, daiDen);
+            if (SafeMath.add(totalDaiSpent, daiAmount) > globalDaiCap) {
+                return false;
+            }
+            totalDaiSpent = SafeMath.add(totalDaiSpent, daiAmount);
+        }
+
+        return true;
+    }
+
+    function isPeriodOver()
+        internal
+        returns (bool)
+    {
+        if (rolling && getNow() > lastStartTime + timePeriod) {
+            lastStartTime = getNow();
+            return true;
+        } else if (!rolling && currentStartTime() > lastStartTime) {
+            lastStartTime = currentStartTime();
+            return true;
+        }
+
+        return false;
+    }
+
+    // Return now from function to allow mocking.
+    function getNow()
```

Comment:
hmm not the biggest fan of that, since this adds unnecessary gas costs. To change "now" during tests you could use @digix/tempo (more info here https://ethereum.stackexchange.com/questions/15755/simulating-the-passage-of-time-with-testrpc)